------------------------------------------------------------------------
r2528 | lyonel | 2013-08-24 18:12:05 +0200 (lø., 24 aug. 2013) | 2 lines

fix #632: ensure that FAT superblock is at least 1 sector long


Index: src/core/fat.cc
===================================================================
--- src/core/fat.cc	(revisjon 2527)
+++ src/core/fat.cc	(revisjon 2528)
@@ -81,6 +81,7 @@
 			uint8_t dummy2[164];
 			uint8_t pmagic[2];
 		} __attribute__((__packed__)) fat32;
+		char sector[512];	// to make sure the whole struct is at least 512 bytes long
 	} __attribute__((__packed__)) type;
 } __attribute__((__packed__));
 

------------------------------------------------------------------------
